src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +281,31 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		/*
+		 * Move this script to the footer if:
+		 * 1. The script is in the header group.
+		 * 2. The current output is the header.
+		 * 3. The intended strategy is delayed.
+		 * 4. The actual strategy is not delayed.
+		 * 5. All dependent scripts are in the footer.
+		 */
+		if ( $intended_strategy && ! $this->is_delayed_strategy( $strategy ) && 0 === $this->groups[ $handle ] && 0 === $group ) {
+			// Check that all depedencies are in the footer.
+			$all_deps_in_footer = true;
+			foreach ( $this->get_dependents( $handle ) as $dep ) {
+				// Exit early if any dependency is not in the footer.
+				if ( isset( $this->groups[ $dep ] ) && 0 === $this->groups[ $dep ] ) {
+					$all_deps_in_footer = false;
+					break;
+				}
+			}
+			if ( $all_deps_in_footer ) {
```

Comment:
Maybe this can be made a bit cleaner if the logic were put into a `private` method? I can amend the PR with what I have in mind if you like.

Reply:
There. How about 1e75fd71f0?

Reply:
Nice, thanks!